### PR TITLE
Pyscal solid liquid

### PR DESCRIPTION
**Pull request by Leimeroth**
Adds pyscals ability to find solid and liquid atoms in a structure. 
Also adds a get_system convenience function to have easier access to pyscal functionality that is not directly accessible and replaces repetitive code with the get_system function.

### CODE REVIEW

File: pyiron_atomistics/atomistics/structure/pyscal.py
Comment (pmrv) [inline pyiron_atomistics/atomistics/structure/pyscal.py]: I think for consistency we would want to call that `pyiron_to_pyscal`, similar to `pyiron_to_ase` and `pyiron_to_pymatgen`.

Comment (Leimeroth) [reply pyiron_atomistics/atomistics/structure/pyscal.py]: renamed it. I also moved it next to the to_ase, to_pymatgen and to_ovito methods

File: pyiron_atomistics/atomistics/structure/pyscal.py
Comment (pmrv) [inline pyiron_atomistics/atomistics/structure/pyscal.py:250]: This returns the number of *atoms* that are inside a solid phase?  If so it'd cool to also return an index array, that shows which are solid and which are liquid, maybe behind a switch, like the `pyscal_adaptive_cna` method.

File: pyiron_atomistics/atomistics/structure/analyse.py
Comment (pmrv) [inline pyiron_atomistics/atomistics/structure/analyse.py:542]: Typo?

File: pyiron_atomistics/atomistics/structure/pyscal.py
Comment (Leimeroth) [reply pyiron_atomistics/atomistics/structure/pyscal.py:250]: This only returns the number by default, but when using return_system=True it returns the complete pyscal system object, which allows to get the indices. I thought this is the most simple in the default case and the most flexible approach when using the return_system option.
Alternatively I could change the default to return an array with the indices of all atoms that are identified as solid, so that the len(returned_array) gives the number of solids I guess. But I do not know if the order of the atoms stays the same when converting between pyiron ase and pyscal system, so I am not sure if indices will always be correct

File: pyiron_atomistics/atomistics/structure/analyse.py
Comment (Leimeroth) [reply pyiron_atomistics/atomistics/structure/analyse.py:542]: what do you mean?

File: pyiron_atomistics/atomistics/structure/pyscal.py
Comment (pmrv) [reply pyiron_atomistics/atomistics/structure/pyscal.py:250]: Ah, I had simply missed that while skimming.  Since anything might happen while changing from pyiron to pyscal returning the full system is ok, I suppose.

File: pyiron_atomistics/atomistics/structure/analyse.py
Comment (pmrv) [reply pyiron_atomistics/atomistics/structure/analyse.py:542]: Since I misunderstood the method, I thought `or` should be `of` or some such...  I promise to read more carefully next time. ;)

**Comment by Leimeroth** [comment]
I think the tests are failing due to #409, since the error messages seem unrelated to my changes. I will wait till this is sorted out and than merge this when tests run, if you don't see any other problems @pmrv.

**Comment by pmrv** [comment]
Yep, once the h5io is sorted out, go ahead and merge.

**Review by pmrv — APPROVED** [review]

**Comment by niklassiemer** [comment]
I fixed the h5io version (for now) in #410, thus, try to rerun the tests :)

**Comment by liamhuber** [comment]
Cool! We are really getting a lot of pyscal mapped functions -- what about doing a refactor (after this PR, not part of it) so that we nest these bindings, i.e. `structure.analyse.pyscal.find_solids` instead of `structure.analyse.pyscal_find_solids`?

**Comment by Leimeroth** [comment]
> Cool! We are really getting a lot of pyscal mapped functions -- what about doing a refactor (after this PR, not part of it) so that we nest these bindings, i.e. `structure.analyse.pyscal.find_solids` instead of `structure.analyse.pyscal_find_solids`?

I always tried to type exactly that.

**Pull request merged**
